### ISSUE TITLE
Return data from sandbox to background

### ISSUE THREAD

**Issue by bestplay9384** [gajananpp/puppeteer-ide-extension#5]
Hey,

As far as i understand from source only CDP Events are passed to background from sandbox. Is there any way to return some gathered data to background then? 

I'm trying to make an extension that can be invoked by externalMessage so it can be received in background only. That is why i'm trying to find out if result of puppeteer script can be passed with simple return, other than passing data via console.log or something :)

**Comment by gajananpp** [comment]
Hi @bestplay9384

You can directly execute puppeteer script in background using [puppeteer-extension-transport](https://github.com/gajananpp/puppeteer-extension-transport) package.

Checkout this example 
https://github.com/gajananpp/puppeteer-extension-transport/tree/main/examples/extension-v3

**Comment by bestplay9384** [comment]
Hey @gajananpp,

I can't. I've external puppeteer scripts for different websites, i'm fetching one and needs to eval it. There is no possibility to `eval` in background since manifest v3 :/

**Comment by gajananpp** [comment]
@bestplay9384, yes creating or migrating to v3 extension is really difficult if it involves working with `eval`. I thought about your use case but couldn't think of any direct way to send data back to background. I hope at least they would provide `chrome.runtime.sendMessage` api in sandbox page in future. For debugging purpose in this extension, i am storing all the CDP commands and their responses in an object in background page, maybe something like that could be done if only response has to be gathered.

[https://github.com/gajananpp/puppeteer-ide-extension/blob/main/src/background.ts#L129-L133](https://github.com/gajananpp/puppeteer-ide-extension/blob/main/src/background.ts#L129-L133)
